json_encoder.go

```diff
@@ -168,6 +172,52 @@ func (enc *jsonEncoder) AddMarshaler(key string, obj LogMarshaler) error {
 	return err
 }
 
+func (enc *jsonEncoder) arrayBegin() {
+	enc.bytes = append(enc.bytes, '[')
+}
+
+func (enc *jsonEncoder) arrayEnd() {
+	enc.bytes = append(enc.bytes, ']')
+}
+
+func (enc *jsonEncoder) arraySep() {
+	enc.bytes = append(enc.bytes, ',')
+}
+
+// AddInts adds a string key and integer slice to the encoder's fields. The key
+// is JSON-escaped.
+func (enc *jsonEncoder) AddInts(key string, vals []int) {
+	enc.addKey(key)
+	enc.arrayBegin()
+	l := len(vals)
+	if l > 0 {
+		enc.bytes = strconv.AppendInt(enc.bytes, int64(vals[0]), 10)
+
+		for i := 1; i < l; i++ {
+			enc.arraySep()
+			enc.bytes = strconv.AppendInt(enc.bytes, int64(vals[i]), 10)
+		}
+	}
+	enc.arrayEnd()
+}
+
+// AddStrings adds a string key and string slice to the encoder's fields. The key
+// is JSON-escaped.
+func (enc *jsonEncoder) AddStrings(key string, vals []string) {
+	enc.addKey(key)
+	enc.arrayBegin()
+	l := len(vals)
+	if l > 0 {
+		enc.wrapString(vals[0])
+
+		for i := 1; i < l; i++ {
+			enc.arraySep()
+			enc.wrapString(vals[i])
+		}
+	}
+	enc.arrayEnd()
+}
```

Comment:
I've reverted the loop unrolling micro-op in favor of a cleaner range iteration and abstracted the separator encoding decision into the encoder's addArraySep(i int) function.  This was done for both JSON and Text encoders.